aws_advanced_python_wrapper/limitless_connection_plugin.py

```diff
@@ -0,0 +1,543 @@
+        logger.debug("LimitlessRouterMonitor.Stopped", self.host_info.host)
+
+    def _get_current_time(self):
+        return datetime.now().microsecond / 1000  # milliseconds
+
+    def run(self):
+        logger.debug("LimitlessRouterMonitor.Running", self.host_info.host)
+
+        context: TelemetryContext = self._telemetry_factory.open_telemetry_context(
+            "limitless router monitor thread", TelemetryTraceLevel.TOP_LEVEL)
+        if context is not None:
+            context.set_attribute("url", self._host_info.url)
+
+        try:
+            while not self.is_stopped:
+                self._open_connection()
+
+                if self._monitoring_conn is not None:
+
+                    new_limitless_routers = self._query_helper.query_for_limitless_routers(self._monitoring_conn, self._host_info.port)
+                    self._limitless_router_cache.compute_if_absent(self._limitless_router_cache_key,
+                                                                   lambda _: new_limitless_routers,
+                                                                   WrapperProperties.MONITOR_DISPOSAL_TIME_MS.get(self._properties))
+
+                sleep(self._interval_ms / 1000)
+
+        except InterruptedError:
+            logger.debug("LimitlessRouterMonitor.InterruptedExceptionDuringMonitoring", self.host_info.host)
+        except Exception:
+            # this should not be reached; log and exit thread
+            logger.debug("LimitlessRouterMonitor.ExceptionDuringMonitoringStop", self.host_info.host)
+
+        finally:
+            self._is_stopped.set()
+            if self._monitoring_conn is not None:
+                try:
+                    self._monitoring_conn.close()
+                except Exception:
+                    # Ignore
+                    pass
+
+            if context is not None:
+                context.close_context()
+
+    def _open_connection(self):
+        try:
+            driver_dialect = self._plugin_service.driver_dialect
+            if self._monitoring_conn is None or driver_dialect.is_closed(self._monitoring_conn):
+                logger.debug("LimitlessRouterMonitor.OpeningConnection", self.host_info.url)
+                self._monitoring_conn = self._plugin_service.force_connect(self._host_info, self._properties, None)
+                logger.debug("LimitlessRouterMonitor.OpenedConnection", self._monitoring_conn)
+
+        except Exception:
+            if self._monitoring_conn is not None:
+                try:
+                    self._monitoring_conn.close()
+                except Exception:
+                    pass  # ignore
+
+                    self._monitoring_conn = None
+
+
+class LimitlessQueryHelper:
+
+    _DEFAULT_QUERY_TIMEOUT_MS: int = 5000
+
+    def __init__(self, plugin_service: PluginService):
+        self._plugin_service = plugin_service
+
+    def query_for_limitless_routers(self, connection: Connection, host_post_to_map: int) -> List[HostInfo]:
```

Comment:
Typo that needs to be updated here and in the rest of the file:
```suggestion
    def query_for_limitless_routers(self, connection: Connection, host_port_to_map: int) -> List[HostInfo]:
```